dev-lang/zig/zig-0.13.0-r1.ebuild

```
@@ -0,0 +1,242 @@
+SLOT="${ZIG_SLOT}"
+IUSE="doc +llvm test"
+RESTRICT="!test? ( test )"
+REQUIRED_USE="
+	!llvm? ( !doc )
+	llvm? ( ${LLVM_REQUIRED_USE} )
+"
+
+# Used by both cmake and zig-build eclass.
+BUILD_DIR="${WORKDIR}/${P}_build"
+
+# Zig requires zstd and zlib compression support in LLVM, if using LLVM backend.
+# (non-LLVM backends don't require these)
+# They are not required "on their own", so please don't add them here.
+# You can check https://github.com/ziglang/zig-bootstrap in future, to see
+# options that are passed to LLVM CMake building (excluding "static" ofc).
+DEPEND="
+	llvm? (
+		$(llvm_gen_dep '
+			sys-devel/clang:${LLVM_SLOT}
+			sys-devel/lld:${LLVM_SLOT}
+			sys-devel/llvm:${LLVM_SLOT}[zstd]
+		')
+	)
+"
+BDEPEND+="
+	test? ( !!<sys-apps/sandbox-2.39 )
+	${DEPEND}
+"
+RDEPEND="${DEPEND}"
+IDEPEND="app-eselect/eselect-zig"
+
+DOCS=("README.md" "doc/build.zig.zon.md")
+
+PATCHES=(
+	"${FILESDIR}/zig-0.13.0-test-std-kernel-version.patch"
+	"${FILESDIR}/zig-0.13.0-skip-test-stack_iterator.patch"
+)
+
+# zig-build does not set this for us since we use ZIG_OPTIONAL=1
+QA_FLAGS_IGNORED="usr/.*/zig/${PV}/bin/zig"
+
+# Since commit https://github.com/ziglang/zig/commit/e7d28344fa3ee81d6ad7ca5ce1f83d50d8502118
+# Zig uses self-hosted compiler only
+CHECKREQS_MEMORY="4G"
+
+pkg_setup() {
+	# Skip detecting zig executable.
+	ZIG_EXE="not-applicable" ZIG_VER="${PV}" zig-build_pkg_setup
+
+	export ZIG_SYS_INSTALL_DEST="${EPREFIX}/usr/$(get_libdir)/zig/${PV}"
+
+	if use llvm; then
+		tc-is-cross-compiler && die "USE=llvm is not yet supported when cross-compiling"
+		llvm-r1_pkg_setup
+	fi
+
+	check-reqs_pkg_setup
+}
+
+src_unpack() {
+	if [[ ${PV} == 9999 ]]; then
+		git-r3_src_unpack
+	else
+		verify-sig_src_unpack
+	fi
+}
+
+src_prepare() {
+	# Avoid double patching. Copied from net-misc/sunshine-0.23.1::gentoo .
+	default_src_prepare
+	default_src_prepare() { :; }
+
+	use llvm && cmake_src_prepare
+	zig-build_src_prepare
+
+	# Remove "limit memory usage" flags, it's already verified by
+	# CHECKREQS_MEMORY and causes unneccessary errors. Upstream set them
+	# according to CI OOM failures, which are not applicable to normal Gentoo build.
+	sed -i -e '/\.max_rss = .*,/d' build.zig || die
+}
+
+src_configure() {
+	# Has no effect on final binary and only causes failures during bootstrapping.
+	filter-lto
+
+	# Used during bootstrapping. stage1/stage2 have limited functionality
+	# and can't resolve native target, so we pass target in exact form.
+	declare -r -g ZIG_HOST_AS_TARGET=$(zig-toolchain_get_target ${CBUILD:-${CHOST}})
+
+	# Note that if we are building with CMake, "my_zbs_args"
+	# are used only after compiling zig2.
+	local my_zbs_args=(
+		--zig-lib-dir "${S}/lib/"
+		# Will be a subdir under ZIG_SYS_INSTALL_DEST.
+		--prefix-lib-dir lib/
+
+		# These are built separately
+		-Dno-langref
+		-Dstd-docs=false
+
+		--release=fast
+	)
+	if use llvm; then
+		my_zbs_args+=(
```

The convention is usually no underscores, but I don't insist:
```suggestion
		myzbcargs+=(
```

---

I personally find it less readable, pronouncable and inconsistent with other local variables and function names tthis way